eclair-core/src/main/scala/fr/acinq/eclair/db/BackupHandler.scala

```diff
@@ -0,0 +1,87 @@
+package fr.acinq.eclair.db
+
+import java.io.File
+
+import akka.actor.{Actor, ActorLogging, Status}
+import akka.pattern.pipe
+import fr.acinq.eclair.channel.ChannelPersisted
+import grizzled.slf4j.{Logger, Logging}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class BackupHandler(databases: Databases, wip: File, destination: File)(implicit ec: ExecutionContext) extends Actor with ActorLogging {
+  import BackupHandler._
+
+  // we listen to ChannelPersisted events, which will trigger a backup
+  override def preStart(): Unit = {
+    super.preStart()
+    context.system.eventStream.subscribe(self, classOf[ChannelPersisted])
+  }
+
+  override def unhandled(message: Any): Unit = {
+    super.unhandled(message)
+    log.warning(s"unhandled message $message")
+  }
+
+  def receive = idle
+
+  // idle mode: start backup as soon as we receive an event
+  def idle: Receive = {
+    case persisted: ChannelPersisted =>
+      doBackup(databases, wip, destination, persisted) pipeTo self
+      context become busy(None)
+  }
+
+  // busy mode: there is already a backup in progress
+  // `again` tells us if we have to do another one when we're done with the current one
+  def busy(again: Option[ChannelPersisted]): Receive = {
+    case persisted: ChannelPersisted =>
+      log.info(s"replacing pending backup with a newer one")
+      context become busy(Some(persisted))
+
+    case Done if again.isDefined =>
```

Comment:
I thought we agree to stop or at least limit doing several handlers for the same message? How about:

```scala
case Done =>
  again match {
    case Some(persisted) =>  
      doBackup(databases, wip, destination, again.get) pipeTo self
      context become busy(None)
    case None =>
      context become idle
```